src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrCodeflower.cs

```diff
@@ -454,6 +478,139 @@ private async Task<ForwardFlow> GetLastForwardFlow(string mappingName)
+                .Select(f => f.Substring(prefix.Length + 1))];
+        }
+
+        return changedFiles;
+    }
+
+    /// <summary>
+    /// Handles files that were reverted in the source repository and would not be included in the PR branch changes.
+    /// Works around a problem that happens in the following scenario:
+    /// 
+    ///   repo                   VMR
+    ///     O───────────────────►O 0. 
+    ///     │                 2. │
+    ///   1.O────────────────O   │
+    ///     │                │   │
+    ///     │                └──►O 3.
+    ///     │                    │
+    ///   4.O─────────────────x  │
+    ///     │                 5. │
+    ///
+    /// The following happens:
+    ///    0. Repo and VMR are initialized
+    ///    1. Two files(`conflict.txt` and `revert.txt`) are added in the repo
+    ///    2. FF is opened and in the PR branch, we change `conflict.txt` to something
+    ///    3. FF PR is merged
+    ///    4. The `revert.txt` file is reverted in the repo, the `conflict.txt` is changed to something else
+    ///    5. The next forward flow will conflict over the `conflict.txt` file
+    ///       This means the FF branch will be based on 0.
+    ///       This means the FF branch needs to have all the changes from the repo(1-4)
+    ///       BUT the revert.txt won't be part of the changes because it was reverted
+    ///       That means the PR branch won't remove it and it will stay in the VMR (even after we resolve the conflict)
+    ///
+    /// This method detects such reverts and resets the file so they match the source repository.
+    /// </summary>
+    private async Task HandleRevertedFiles(
+        SourceMapping mapping,
+        ILocalGitRepo repo,
+        Codeflow currentFlow,
+        string shaBeforeRecreation,
+        IReadOnlyCollection<string> changedFilesAfterRecreation,
+        CancellationToken cancellationToken)
+    {
+        bool currentIsBackflow = currentFlow is Backflow;
+        var changedFilesAfterCurrentChanges = await GetChangesInHeadBranch(
+            mapping,
+            repo,
+            currentIsBackflow,
+            shaBeforeRecreation,
+            cancellationToken);
+
+        // We compare changed files before and after current changes to see if a file was reverted in between
+        // Such file would not appear in the PR branch changes, so we'd need to handle it ourselves
+        var revertedFiles = changedFilesAfterRecreation
+            .Where(f => !changedFilesAfterCurrentChanges.Contains(f))
+            .ToList();
+
+        if (revertedFiles.Count == 0)
+        {
+            return;
+        }
+
+        _logger.LogInformation("Reverted files detected after applying changes: {files}. Resetting the files to their current state.",
+            string.Join(", ", revertedFiles));
+
+        string vmrPrefix = VmrInfo.GetRelativeRepoSourcesPath(mapping.Name).ToString();
```

Comment:
why convert this to a string? It's a unix path so when we use it bellow it should turn out as we want it to

Reply:
```suggestion
        UnixPath vmrPrefix = VmrInfo.GetRelativeRepoSourcesPath(mapping.Name);
```

Reply:
we won't need the string interpolation on lines 574 and 577 too I think. The `/` operation that we have should give us the same result with a UnixPath